extensions/blocks/amazon/edit.js

```diff
@@ -1,23 +1,248 @@
+						<ToggleControl
+							label={ __( 'Show Image', 'jetpack' ) }
+							checked={ showImage }
+							onChange={ () => setAttributes( { showImage: ! showImage } ) }
+						/>
+						<ToggleControl
+							label={ __( 'Show Title', 'jetpack' ) }
+							checked={ showTitle }
+							onChange={ () => setAttributes( { showTitle: ! showTitle } ) }
+						/>
+						<ToggleControl
+							label={ __( 'Show Author/Seller', 'jetpack' ) }
+							checked={ showSeller }
+							onChange={ () => setAttributes( { showSeller: ! showSeller } ) }
+						/>
+						<ToggleControl
+							label={ __( 'Show Price', 'jetpack' ) }
+							checked={ showPrice }
+							onChange={ () => setAttributes( { showPrice: ! showPrice } ) }
+						/>
+						<ToggleControl
+							label={ __( 'Show Purchase Button', 'jetpack' ) }
+							checked={ showPurchaseButton }
+							onChange={ () => setAttributes( { showPurchaseButton: ! showPurchaseButton } ) }
+						/>
+					</PanelBody>
+					<PanelColorSettings
+						title={ __( 'Color Settings', 'jetpack' ) }
+						colorSettings={ [
+							{
+								value: backgroundColor,
+								onChange: newBackgroundColor =>
+									setAttributes( { backgroundColor: newBackgroundColor } ),
+								label: __( 'Background Color', 'jetpack' ),
+							},
+							{
+								value: textColor,
+								onChange: newTextColor => setAttributes( { textColor: newTextColor } ),
+								label: __( 'Text Color', 'jetpack' ),
+							},
+							{
+								value: buttonAndLinkColor,
+								onChange: newButtonAndLinkColor =>
+									setAttributes( { buttonAndLinkColor: newButtonAndLinkColor } ),
+								label: __( 'Button & Link Color', 'jetpack' ),
+							},
+						] }
+					>
+						{
+							<ContrastChecker
+								{ ...{
+									isLargeText: false,
+									textColor: textColor,
+									backgroundColor: backgroundColor,
+								} }
+							/>
+						}
+					</PanelColorSettings>
+				</>
+			) }
+		</InspectorControls>
+	);
 
-export default AmazonEdit;
+	const blockPreview = () => {
+		const {
+			title,
+			detailPageUrl,
+			listPrice,
+			imageUrlMedium,
+			imageWidthMedium,
+			imageHeightMedium,
+		} = data.products.filter( productDataItem => productDataItem.asin === asin )[ 0 ];
+
+		// TODO - we should be able to get this from API in a neater way once we have access
+		const seller = 'TODO';
+
+		// TODO - we have different image sizes in the API
+		const image = imageUrlMedium && (
+			<a href={ detailPageUrl }>
+				<img
+					alt={ title }
+					src={ imageUrlMedium }
+					width={ imageWidthMedium }
+					heigth={ imageHeightMedium }
+				/>
+			</a>
+		);
+
+		const buttonTextColor = tinycolor
+			.mostReadable( buttonAndLinkColor, [ '#ffffff' ], {
+				includeFallbackColors: true,
+				size: 'small',
+			} )
+			.toHexString();
+
+		if ( ! asin ) {
+			return null;
+		}
+
+		return (
+			<div
+				style={ { backgroundColor: backgroundColor, color: textColor, width: imageWidthMedium } }
```

Comment:
Should we be using `withColors` and allowing the colours to be set with the various `has-primary-color` classes that the theme should provide?

Reply:
Good idea. Added to the list!